### DISCUSSION TITLE
Suggestions and a license question

### DISCUSSION THREAD

**Discussion by julia-and** [nathanwoulfe/Plumber-2#121] [Q&A] [answered]
We are evaluating Plumber 2 for a client project and have noticed a few things / have a few questions:

1. We would like to make comments a required field when the reviewer is Rejecting. Is this possible currently, or could it be added as an option?
2. After a workflow completes, the content tree should be refreshed to reflect the updated content
3. [minor] If, as an admin, you Unpublish (bypassing workflow), the Publish button doesn't show the Workflow options any more:
![image](https://user-images.githubusercontent.com/20821578/156837511-7ab70715-2ad7-42ab-b1d8-31bcff4b234b.png)
4. Do we need multiple licenses if we are running a multi-site instance with different TLDs?

Thanks for this excellent product!

**Comment by nathanwoulfe** [comment]
Hi @sytexa-julia, hopefully the below helps...

1/ That sounds like a bug - the comment should be a mandatory field for all other than the final stage. Will fix that in the next patch release
2/ Do you mean the icon in the content tree isn't updating from showing saved content? Also sounds like a bug, I've been seeing issues with signalR not always behaving the way it should, so need to review how changes are refreshed (probably using normal AngularJs events instead)
3/ I think that's intentional, but doesn't make sense - an unpublished node should be able to send for [re]-publishing. Does the missing button return after refreshing the page?
4/ The license is bound to the domain serving the backoffice, which shouldn't change for a multi-site install (I think? Haven't done much in that space). If you had 10 sites in a single Umbraco instance, all edited from mydomain.com/umbraco, one license do the trick. If they're accessed from different domains, I don't actually know how Plumber would work...

**Comment by julia-and** [comment]
For 2/, my test was this:
1. Create a new content item, save and request publish. In the tree, the node's text is colored grey instead of black, indicating it is not published yet.
2. As admin, approve the item through the workflow. The item is still colored grey in the tree. 

I've seen intermittent cases where other bits of backoffice state related to the workflow aren't updated--could well be related to the signalR issue you mentioned. Hard to pin this down.

For 3/, yes, after refreshing the missing button returns.

Something else the client expressed interest in, and might make sense to add to Plumber--show the Schedule Publish / Unpublish fields on the Initiate Workflow screen and make them optionally required by document type.

Thank you for the clarification on the other items!

**Comment by nathanwoulfe** [comment]
Hi @sytexa-julia, I'll have fixes for 1 and 3 in the next patch release. For 3, it's a little tricky in some scenarios (and Umbraco has the same issue), something like this:
- load unpublished node
- user buttons won't have unpublish as that's not valid for the unpublished node
- publish node
- no unpublish button available until the node is reloaded

I've improved how Plumber sets the buttons, but there are still some cases (like the above), where the unpublish button will be missing because it didn't exist when the node was loaded, so can't be displayed after publishing.

Re the scheduling fields, that is available via the scheduled date property when initiating a workflow. This updates Umbraco's content schedule, with the [un]publish occurring either after the workflow completes and the scheduled date passes, or immediately on workflow completion if the scheduled date has already passed.
![image](https://user-images.githubusercontent.com/3248070/157795279-2541b392-b6df-4841-9b0f-217cfbaeee58.png)

**Reply by julia-and** [reply]
Great stuff, thanks for taking care of those bugs! 

Re the scheduling fields, we're looking for the ability to set both the Publish and Unpublish dates from the same screen. The client is thinking they want to require their editors to set an Unpublish date whenever they publish, to keep cruft from building up (their current production site is 80%+ outdated articles / content). 

Would something like that be possible?

**Reply by nathanwoulfe** [reply]
That's a bit more difficult, given Plumber separates publish and unpublish workflows. I'm not sure about the UX of being able to set an unpublish date in a publish workflow, as that would allow a publish workflow to set an unpublish date in the near future, and effectively work as an unpublish workflow... I'll have a think about it, it's technically possible, just not sure on the usability.

Alternative for the client could be to have a max-age on different documents, and a scheduled task to unpublish items. That way there's no requirement for editors, it just happens automatically.

**Comment by nathanwoulfe** [comment, marked as answer]
@sytexa-julia just a quick FYI - planning on releasing 2.1.4 next week with these fixes and some general behaviour improvements (aka less use of signalR)
